src/state.py

```diff
@@ -11,6 +12,55 @@ def __init__(self):
     def add_student(self, student):
         self.students[student.student_id] = student
 
+    def get_score(self): 
+        score = 0
+        
+        for student in self.students.values(): 
+
+            #checking buddies 
+            for numbers in set(list(student.buddies.values())):
+                    for a_class in set(list(student.buddies.keys())): 
+                        if student.subjects_and_classes[a_class] == self.students[numbers].subjects_and_classes[a_class]: 
+                            score += 30         #the person is at the same class of his friend
+                            student.alone = False
+                        else: 
+                            score -= 20
+                        
+            #checking if student got a target class
+            for position, j in enumerate(student.subjects_and_classes): 
+                if j in student.subject_targets.keys(): 
+                    if student.subjects_and_classes[j] in student.subject_targets[j]:  
+                        score += 40    
+                        student.target = True 
+                    else: 
+                        score -= 30
+
+                #checking if a student gave in any classes 
+                if j in student.subject_give_ins.keys():
+                    if student.subjects_and_classes[j] in student.subject_give_ins[j]: 
+                        score -= 3
+                        student.gave_in = True
+                    else: 
+                        score += 5
+                     
+
+               #checking for schedule conflicts.
+                for p in range(position+1, len(list(student.subjects_and_classes.keys()))):
+                    key = list(student.subjects_and_classes.keys())[p] 
+                    sched_1 = self.class_schedules[translate_subject_and_class(j,student.subjects_and_classes[j])]
+                    sched_2 = self.class_schedules[translate_subject_and_class(key, student.subjects_and_classes[key])] 
+                    if (sched_1.conflicts(sched_2)):
+                        return float('-inf')
+
+            # If he gives up a class but didn't get the target nor he is with his buddy
+            if student.gave_in and student.target and student.alone and list(student.buddies): 
```

Comment:
Shouldn't this condition have **and not student.target**?
Also, why do you have a list as part of the condition?
Same question goes for line 59

Reply:
God, you are right!

Reply:
The second is redundant and can be refactored! I will fix it.